ex1/dates_times.py

```diff
@@ -32,13 +32,89 @@
 
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
-
+from datetime import date
+def last_of_month(dt):
+    """
+    takes an argument dt of type date and returns a date object representing
+    the last day of the month dt was in (takes into consideration leap years)
+    Parameters
+    ----------
+    dt: date
+        determines month
+    Returns
+    -------
+    lday: date
+        last day of this month
+    """
+    short = [4, 6, 9, 11]
+    if dt.month in short:
+        day = 30
+    elif dt.month == 2:
+        if dt.year % 400 == 0 or dt.year % 4 == 0 and dt.year % 100 != 0:
+            day = 29
+        else:
+            day = 28
```

Comment:
It's not necessary to implement this yourself. The `calendar.monthrange` function does this for you. See #20